### PR TITLE
Make test query optional

### PR DESCRIPTION
**Pull request by pkoenig10**
Small FLUP to https://github.com/palantir/atlasdb/pull/6499 that more accurately represents the intention of the test query config by using `Optional` rather than an empty string to indicate no test query.

### CODE REVIEW

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/ConnectionConfig.java [hunk 1]
```diff
@@ -58,10 +58,7 @@ public MaskedValue getDbPassword() {
 
     public abstract String getDriverClass();
 
-    @Value.Default
-    public String getTestQuery() {
-        return "";
-    }
+    public abstract Optional<String> getTestQuery();
```
Comment (jeremyk-91) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/ConnectionConfig.java:61]: This will cause an API break 🔥 though an internal code search found that the only user of this is our internal implementation of the timelock service. I think we can accept this break, but @Sam-Kramer we'll likely need to switch that to deal correctly with this type change next week

**Review by jeremyk-91 — APPROVED** [review]
Agree this is cleaner - there's a small snag around API breaks, but I think we can work around that
